django/contrib/gis/geos/mutable_list.py

```diff
@@ -272,7 +272,7 @@ def _assign_extended_slice_rebuild(self, start, stop, step, valueList):
 
         # we're not changing the length of the sequence
         newLen = len(self)
-        newVals = dict(zip(indexList, valueList))
+        newVals = dict(zip(indexList, valueList, strict=True))
```

Comment:
> Do we need to add strict=True to all zip() calls? This seems unnecessary in many cases.

I think that the benefit is that it prevents any inadvertent regressions. Without it, in most cases, there is no check and so if one of the arguments is modified, e.g. an extra value added for list in a test case, it may not actually get handled as expected.

> `indexList` and `valueList` cannot have different lengths, what is the reason for adding `strict=True`? 🤔

I guess in these cases an explicit check was added above so there isn't all that much benefit here aside from preventing potential regressions. _(In an ideal world `zip()` would have always had strict length checking.)_

Reply:
Also, in a similar vein to adding explicit `strict=False`, it signals what the expected behavior is.

Reply:
@felixxm I've added a fixup that reverts these two cases which you can keep or remove depending on what you decide here. I still think there is value, as explained above.